apps/api/src/billing/services/chain-error/chain-error.service.spec.ts

```diff
@@ -0,0 +1,126 @@
+import type { BalanceHttpService } from "@akashnetwork/http-sdk";
+import type { EncodeObject } from "@cosmjs/proto-signing";
+import type { MockProxy } from "jest-mock-extended";
+import { mock } from "jest-mock-extended";
+
+import type { Wallet } from "@src/billing/lib/wallet/wallet";
+import type { BillingConfigService } from "@src/billing/services/billing-config/billing-config.service";
+import { ChainErrorService } from "./chain-error.service";
+
+const USDC_IBC_DENOMS = {
+  mainnetId: "ibc/170C677610AC31DF0904FFE09CD3B5C657492170E7E52372E48756B71E56F2F1",
+  sandboxId: "ibc/12C6A0C374171B595A0A9E18B83FA09D295FB1F2D8C6DAA3AC28683471752D84"
+};
+
+describe(ChainErrorService.name, () => {
+  describe("toAppError", () => {
+    const encodeMessages: EncodeObject[] = [];
+
+    it("returns the original Error when no clue is found", async () => {
+      const { service } = setup();
+      const err = new Error("just some random failure");
+      const result = await service.toAppError(err, encodeMessages);
+      expect(result).toBe(err);
+    });
+
+    it("returns 503 when balance < required for uakt", async () => {
+      const { service, balanceHttpService } = setup();
+      const denom = "uakt";
+      const err = new Error(`insufficient funds: 10${denom} is smaller than 20${denom}`);
+      balanceHttpService.getBalance.mockResolvedValue({ amount: 5, denom: "uakt" });
+
+      const appErr = (await service.toAppError(err, encodeMessages)) as any;
```

Comment:
pls avoid `any`s. This case could be handled like so:
```suggestion
      import { ServiceUnavailable } from "http-errors";

      const appErr = await service.toAppError(err, encodeMessages);
      
      expect(appErr).toBeInstanceOf(ServiceUnavailable);
      expect(appErr.message).toBe("Service temporarily unavailable");
```

Reply:
Thanks, updated